joomla/joomla-cms
# [4.0] Fix images

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/35779.

# THIS IS A RELEASE BLOCKER

## Images in many layouts are totally broken

### Summary of Changes
- Use the helper to clean the URL
- Use the HTMLHelper consistently
- Based on the https://github.com/ttc-freebies/plugin-responsive-images/tree/4.0.0/src/plugins/system/responsiveplgoverrides/html that had solved these before 4.0 but somehow I forgot to do a PR here...

### Testing Instructions

Test that images in all these views are rendering correctly.

Create a new contact and use the image field
Create a menu item contact category and check the front end that the image html is not having the hash part
Create a menu item single contact and check the front end that the image html is not having the hash part

Create an article with both intro and full images
Create a menu item articles category blog and check the front end that the image html is not having the hash part
Also check on the article view (clicking on the title) the image html is not having the hash part 

Create a new newsfeed with the link: `https://www.joomla.org/announcements.feed?height=660&keepthis=632&tb_iframe=true&type=rss`, make sure to set both the first and second images
Create a menu item newsfeed category and check the front end that the image html is not having the hash part
Also check the the actual feed has the first and second image with the hash

### Actual result BEFORE applying this Pull Request

Broken

### Expected result AFTER applying this Pull Request

Fixed

### Documentation Changes Required

Not these particular changes but the process of cleaning the URL from the media field should definitely be documented
